src/components/Button/index.tsx

```diff
@@ -71,17 +74,63 @@ export interface ButtonProps extends BaseComponentProps {
    */
   icon?: IconName;
 }
+export interface RegularButtonProps extends BaseButtonProps {
+  /**
+   * Whether this button should be disabled or not. Defaults to false.
+   */
+  disabled?: boolean;
+  /**
+   * Disabled buttons cannot have a Link href
+   */
+  href?: void;
+}
+export interface LinkButtonProps extends BaseButtonProps {
+  /**
+   * If provided, will render a Link styled as a Button
+   */
+  href: string;
+  /**
+   * Link buttons cannot be disabled
+   */
+  disabled?: void;
+}
+export type ButtonProps = RegularButtonProps | LinkButtonProps;
 
 export default class Button extends React.Component<ButtonProps, {}> {
+  static propTypes = {
+    // TypeScript does not support negated types; using PropTypes custom validator instead of runtime validations
+    href (props: LinkButtonProps, propName: string, componentName: string) {
+      const href = props.href;
+      if (typeof href !== 'string') {
+        return;
+      }
+      if (hrefBlacklist.indexOf(href) > -1) {
+        return new Error(
+          `Invalid prop ${propName} supplied to ${componentName}: "${href}".
+          Please use a Button for actions, NavigationLink (or Button with href) for navigation.`,
+        );
+      }
+    },
+  };
+
   static defaultProps = {
     size: ButtonSize.REGULAR,
     color: ButtonColor.PRIMARY,
-    disabled: false,
   };
 
-  public render() {
-    const { color, size, disabled, ariaLabel, text, icon, onClick, onMouseEnter, onMouseLeave, onFocus, onBlur }
-      = this.props;
+  public render () {
+    const { props } = this;
+    const { ariaLabel, text, icon, onClick, onMouseEnter, onMouseLeave, onFocus, onBlur } = props;
+
+    // Only use href for LinkButton, and only use disabled for non-LinkButton
+    let disabled;
+    let href;
+    if ((props as RegularButtonProps).disabled) {
+      disabled = (props as RegularButtonProps).disabled;
+    } else if ((props as LinkButtonProps).href) {
+      href = (props as LinkButtonProps).href;
+    }
```

Comment:
I suspect @benlakey would frown upon this bit of local state. Is there a way we can express this with `const` instead of `let`? I'm totally fine with a ternary operator, but I'm open to other ideas.